ios/FluentUI/Vnext/Drawer/Drawer+UIKit.swift

```diff
@@ -123,18 +120,26 @@ extension MSFDrawer: UIViewControllerTransitioningDelegate, UIViewControllerAnim
 
         state.animationDuration = transitionDuration(using: transitionContext)
 
-        // Delegate animation to swiftui by changing state
+///       The presentation of drawer happens in two steps
+///        1. Present the hosting view (trasnparent) without animation
+///        2. Expand drawer view with animation (depending on the client's preference)
+///        The animation is delegated to swiftUI framework instead of UIKit. The intent of overriding `UIViewControllerTransitioningDelegate` is to provide UIKit client interfaces
         if isPresentingDrawer {
-            transitionContext.containerView.addSubview(drawerView)
-            drawerView.frame = UIScreen.main.bounds
-            if !drawer.isPresentationGestureActive {
-                state.isExpanded = true
-            }
-            transitionContext.completeTransition(!transitionContext.transitionWasCancelled)
+            UIView.animate(withDuration: Constant.disabledAnimationDuration, animations: {
+                transitionContext.containerView.addSubview(drawerView)
+                drawerView.frame = UIScreen.main.bounds
+            }, completion: { [weak self] _ in
+                if let strongSelf = self {
```

Comment:
If nothing else will be done if self is nil, we should you guard let.

https://github.com/microsoft/swift-guide/blob/gh-pages/EarlyReturn.md